frontends/web/src/routes/device/bitbox02/sdcardcheck.tsx

```diff
@@ -14,80 +14,64 @@
  * limitations under the License.
  */
 
-import { Component } from 'react';
-import { translate, TranslateProps } from '../../../decorators/translate';
+import { FunctionComponent, useCallback, useEffect, useState } from 'react';
+import { TranslateProps } from '../../../decorators/translate';
 import { Dialog, DialogButtons } from '../../../components/dialog/dialog';
-import { apiGet } from '../../../utils/request';
 import { Button, ButtonLink } from '../../../components/forms';
+import { checkSDCard } from '../../../api/bitbox02';
+import { useTranslation } from 'react-i18next';
 
-interface SDCardCheckProps {
+type SDCardCheckProps = {
     deviceID: string;
 }
 
-interface State {
-    sdCardInserted?: boolean;
-}
-
-
 type Props = SDCardCheckProps & TranslateProps;
 
-class SDCardCheck extends Component<Props, State> {
-  state = {} as State;
+const SDCardCheck: FunctionComponent<Props> = ({ deviceID, children }) => {
+  const { t } = useTranslation();
+  const [sdCardInserted, setSdCardInserted] = useState<boolean | undefined>();
+  const check = useCallback(() => checkSDCard(deviceID).then(ins => setSdCardInserted(ins)),[deviceID]);
```

Comment:
Just pass the setter?

```diff
diff --git a/frontends/web/src/routes/device/bitbox02/sdcardcheck.tsx b/frontends/web/src/routes/device/bitbox02/sdcardcheck.tsx
index 6f50dfb3..4fb9a2ad 100644
--- a/frontends/web/src/routes/device/bitbox02/sdcardcheck.tsx
+++ b/frontends/web/src/routes/device/bitbox02/sdcardcheck.tsx
@@ -30,7 +30,7 @@ type Props = SDCardCheckProps & TranslateProps;
 const SDCardCheck: FunctionComponent<Props> = ({ deviceID, children }) => {
   const { t } = useTranslation();
   const [sdCardInserted, setSdCardInserted] = useState<boolean | undefined>();
-  const check = useCallback(() => checkSDCard(deviceID).then(ins => setSdCardInserted(ins)),[deviceID]);
+  const check = useCallback(() => checkSDCard(deviceID).then(setSdCardInserted), [deviceID]);
 
   useEffect(() => {
     check();

```

Also small nit, please add space after comma, I'll open a PR with space eslint rule later.